### DISCUSSION TITLE
N8N with Rest API

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#10700] [Help] [answered]
Hi guys,

I'm using the rest API to send emails through n8n.
My goal is to send the emails out as soon as somebody presses the create/update row button.

My problems are: 
- the workflow executes after every single update/change in the form
- executes when pressing a "Navigate to"-Button

Is there any way to fix this? I only need it to execute when pressing a button with the function "Execute query".

Best,
Kas

**Comment by melohagan** [comment]
Hey @kasgithub8 

> the workflow executes after every single update/change in the form

If this is the case, it seems that you are using an *On Change* event to trigger the **Execute query** action?

**Reply by kasgithub8** [reply]
Hi @melohagan,

no, I do not use _On Change_.
I only manually use the **Execute Query** action when pressing the "Create row"- and the "Update row"-Button.

My form has about 14 fields that I need to fill out and after each field it executes, which is wrong.

Sincerely,
Kas

**Reply by melohagan** [reply]
Hey @kasgithub8 I'll need a recording demonstrating this one in action, and possibly an app export if you can provide it, because I can't see why editing the fields in your form would trigger the submit.

**Reply by kasgithub8** [reply]
Hi @melohagan,

the video and export are ready.
Which email do I send it to.

Best,
Kas

**Reply by melohagan** [reply]
Hey @kasgithub8 Please send it through to support@budibase.com

**Reply by kasgithub8** [reply]
@melohagan sent!

**Reply by kasgithub8** [reply]
App export: 

```
[support@budibase.com (support@budibase.com)](mailto:support@budibase.com)
Your message wasn't delivered because the recipient's email provider rejected it.
```

**Reply by melohagan** [reply]
Would you be able to upload it to a file share such as Google Drive, and then email me the link?

**Reply by kasgithub8** [reply]
@melohagan done!

**Reply by melohagan** [reply, marked as answer]
Hey @kasgithub8 Thanks, I've replied to your email. The short answer is that you had a Data provider hooked up to the POST query, which was bound to the form fields.

More here on using bindings in forms, and the difference between POST and GET: https://docs.budibase.com/docs/form-bindings

**Reply by kasgithub8** [reply]
Hi @melohagan,

thanks for the quick answer.
I think I added the data provider at the beginning because I  couldn't use the REST data source in the button and then I just went with it.
Thanks tho, I tried almost everything but that!

**Reply by melohagan** [reply]
No worries, glad to help!
